app/src/main/java/at_challenge/deaguiar/daniel/atchalleng/service/DownloadIntentService.java

```diff
@@ -0,0 +1,123 @@
+package at_challenge.deaguiar.daniel.atchalleng.service;
+
+import android.app.IntentService;
+import android.content.Intent;
+import android.os.Bundle;
+import android.os.ResultReceiver;
+import android.util.Base64;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.DefaultHttpClient;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+/**
+ * DownloadIntentService
+ * Responsible to request data from server
+ *
+ * @author Daniel Besen de Aguiar
+ */
+public class DownloadIntentService extends IntentService {
```

Comment:
This name is not meaningful. I first thought it was downloading a file to the disk or something along these lines.